model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/organization/InfinispanOrganizationProvider.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models.cache.infinispan.organization;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.keycloak.models.IdentityProviderModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.OrganizationModel;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.models.cache.CacheRealmProvider;
+import org.keycloak.models.cache.infinispan.RealmCacheSession;
+import org.keycloak.organization.OrganizationProvider;
+
+public class InfinispanOrganizationProvider implements OrganizationProvider {
+
+    private final KeycloakSession session;
+    private final OrganizationProvider jpaOrgDelegate;
+    private final RealmCacheSession realmCache;
+    private final Map<String, OrganizationAdapter> managedOrganizations = new HashMap<>();
```

Comment:
Can you elaborate why we want to keep adapters in this map? To avoid creating instances all the time?

Reply:
Yeah, other infinispan providers do the same thing to avoid re-creating the adapter in the same session.

Reply:
True, forget about it. I was wondering how much it will actually save in terms of resource usage vs the additional complexity by introducing a session-level cache in front of ISPN.